applications/MultilevelMonteCarloApplication/custom_statistics/power_sums_statistics.cpp

```diff
@@ -0,0 +1,132 @@
+//
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+
+// Application includes
+#include "custom_statistics/power_sums_statistics.h"
+
+
+namespace Kratos
+{
+    /* Public functions *******************************************************/
+
+    // Constructor
+    PowerSumsStatistics::PowerSumsStatistics(
+        ModelPart& rModelPart,
+        Parameters ThisParameters):
+        Process(),
+        mrModelPart(rModelPart)
+    {
+        /**
+         * Configure using the following parameters:
+         * reference_variable_name: variable for which computing the power sums
+         */
+        Parameters default_parameters = Parameters(R"(
+        {
+            "reference_variable_name" : "PLEASE_SPECIFY_VARIABLE_NAME"
+        })"
+        );
+
+        ThisParameters.ValidateAndAssignDefaults(default_parameters);
+
+        // Set parameters
+        mReferenceVariable = ThisParameters["reference_variable_name"].GetString();
+    }
+
+    std::string PowerSumsStatistics::Info() const
+    {
+        return "PowerSumsStatistics";
+    }
+
+    void PowerSumsStatistics::PrintInfo(std::ostream& rOStream) const
+    {
+        rOStream << "PowerSumsStatistics";
+    }
+
+    void PowerSumsStatistics::PrintData(std::ostream& rOStream) const
+    {
+        this->PrintInfo(rOStream);
+    }
+
+    void PowerSumsStatistics::ExecuteInitialize()
+    {
+        KRATOS_TRY;
+
+        auto& r_nodes_array = mrModelPart.Nodes();
+        // Initialize power sums variables
+        VariableUtils().SetNonHistoricalVariableToZero(POWER_SUM_1, r_nodes_array);
+        VariableUtils().SetNonHistoricalVariableToZero(POWER_SUM_2, r_nodes_array);
+        VariableUtils().SetNonHistoricalVariableToZero(POWER_SUM_3, r_nodes_array);
+        VariableUtils().SetNonHistoricalVariableToZero(POWER_SUM_4, r_nodes_array);
+
+        KRATOS_CATCH("");
+    }
+
+    void PowerSumsStatistics::ExecuteFinalizeSolutionStep()
+    {
+        KRATOS_TRY;
+
+        // Check and set number of elements and check dimension
+        KRATOS_ERROR_IF(mrModelPart.NumberOfNodes() == 0) << "The number of nodes in the domain is zero. The power sums statistic cannot be applied."<< std::endl;
+        const unsigned int number_nodes = mrModelPart.NumberOfNodes();
+
+        // Extract informations
+        if (KratosComponents<Variable<double>>::Has(mReferenceVariable)) {
+            const auto& r_reference_var = KratosComponents<Variable<double>>::Get(mReferenceVariable);
+            // Iterate over the nodes
+            #pragma omp parallel for
+            for(int i_node = 0; i_node < static_cast<int>(number_nodes); ++i_node) {
+                auto it_node = mrModelPart.NodesBegin() + i_node;
+                // Retrieve current time step local variable
+                double variable_current = it_node->GetSolutionStepValue(r_reference_var);
+                // Retrieve power sums
+                const double S1_old = it_node->GetValue(POWER_SUM_1);
```

Comment:
you could shorten this a lot by doing

~~~cpp
it_node->GetValue(POWER_SUM_1) += std::pow(variable_current,1);
~~~

Reply:
Done!